src/modules/fornecedor/fornecedor.service.ts

```diff
@@ -0,0 +1,179 @@
+      nome: fornecedor.nome,
+      cnpj: fornecedor.cnpj,
+      email: fornecedor.email,
+      telefone: fornecedor.telefone,
+      endereco: fornecedor.endereco,
+      data_nascimento: format(
+        addDays(new Date(fornecedor.data_nascimento), 1),
+        'dd/MM/yyyy',
+      ),
+    }));
+  }
+
+  async findOne(id: number): Promise<ReturnFornecedorDto> {
+    // Use o Prisma para buscar o fornecedor pelo ID
+    const fornecedor = await this.prisma.fornecedor.findUnique({
+      where: { id },
+    });
+
+    if (!fornecedor) {
+      // Se nenhum fornecedor com o ID especificado for encontrado, lance uma exceção
+      throw new NotFoundException(`Fornecedor com ID #${id} não encontrado`);
+    }
+
+    // Mapeie o fornecedor para o DTO desejado
+    return {
+      nome: fornecedor.nome,
+      cnpj: fornecedor.cnpj,
+      email: fornecedor.email,
+      telefone: fornecedor.telefone,
+      endereco: fornecedor.endereco,
+      data_nascimento: format(fornecedor.data_nascimento, 'dd/MM/yyyy'),
+    };
+  }
+
+  async findByCnpj(cnpj: string): Promise<Fornecedor> {
+    const fornecedor = await this.prisma.fornecedor.findFirst({
+      where: { cnpj },
+    });
+
+    if (!fornecedor) {
+      throw new NotFoundException(`Fornecedor com CNPJ ${cnpj} não encontrado`);
+    }
+
+    return fornecedor;
+  }
+
+  async update(
+    id: number,
+    updateFornecedorDto: UpdateFornecedorDto,
+  ): Promise<Fornecedor> {
+    // Verifique se o fornecedor com o ID especificado existe
+    const existingFornecedor = await this.prisma.fornecedor.findUnique({
+      where: { id },
+    });
+
+    if (!existingFornecedor) {
+      // Se nenhum fornecedor com o ID especificado for encontrado, lance uma exceção
+      throw new NotFoundException(`Fornecedor com ID #${id} não encontrado`);
+    }
+
+    // Use o Class Validator para validar o DTO
+    const errors = await validate(updateFornecedorDto);
+
+    if (errors.length > 0) {
+      // Se houver erros de validação, lance uma exceção com detalhes
+      throw new HttpException(errors, HttpStatus.BAD_REQUEST);
+    }
+
+    // Verifique se o CNPJ ou o email foram modificados
+    if (
+      (updateFornecedorDto.cnpj &&
+        updateFornecedorDto.cnpj !== existingFornecedor.cnpj) ||
+      (updateFornecedorDto.email &&
+        updateFornecedorDto.email !== existingFornecedor.email)
+    ) {
+      // Verifique se já existe alguém com os mesmos campos de CNPJ ou email
+      const duplicateFornecedor = await this.prisma.fornecedor.findFirst({
+        where: {
+          OR: [
+            { cnpj: updateFornecedorDto.cnpj },
+            { email: updateFornecedorDto.email },
+          ],
+        },
+      });
+
+      if (duplicateFornecedor) {
+        // Se já existe alguém com os mesmos campos de CNPJ ou email, lance uma exceção de conflito
+        throw new ConflictException(
+          'Já existe um fornecedor com o mesmo CNPJ ou email',
+        );
+      }
```

Comment:
Verificar se está funcionando, pois se o cnpj for igual e o email não, o fornecedor vai ser encontrado pois continua com o mesmo cnpj